File: demo5/src/main/java/com/example/demo5/MainActivity.java
package com.example.demo5;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private Button mButton;
    private TextView mTv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mButton=findViewById(R.id.button);
        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               startActivity( new Intent(MainActivity.this,Main2Activity.class));
            }
        });
        mTv=findViewById(R.id.tv);
        mTv.setText(isTaskRoot()+"==");
    }
}

File: task1/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.ComponentName
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val task:TextView=findViewById(R.id.taskid1)
        task.text= "task_id==$taskId"
    }
    public fun onTest(view: View){
      val componentName:ComponentName=ComponentName("com.example.task2","com.example.task2.Main2Activity")
        startActivity(Intent().setComponent(componentName))

    }
}
